src/pybind/mgr/cephadm/module.py

```diff
@@ -2948,10 +2948,12 @@ def get_daemon_names(daemons: List[str]) -> List[str]:
                 deps.append(build_url(host=dd.hostname, port=port).lstrip('/'))
             deps = sorted(deps)
         elif daemon_type == 'admin-gateway':
-            # url_prefix for the following services depends on the presence of admin-gateway
-            deps += get_daemon_names(['grafana', 'prometheus', 'alertmanager'])
+            # url_prefix for monitoring daemons depends on the presence of admin-gateway
+            # while dashboard urls depend on the mgr daemons
+            deps += get_daemon_names(['mgr', 'grafana', 'prometheus', 'alertmanager'])
```

Comment:
thank you for highlighting this subtle detail, at this moment `mgmt-gateway` doesn't depend on the active manager (we use the upstream redundancy mechanism to pick the active dashboard). But I'll add a comment related to this behavior just in case somebody in the future modifies the code :+1: